### ISSUE TITLE
'Gather' action is continuous.

### ISSUE THREAD

**Issue by tony-sanguk** [gemblerz/gor-purdue#22]
In SC2, probe can do their actions, but like 'Gather', some actions are continuous. It means that after one cycle to gather minerals, probe doesn't stop to act gathering.

- I think after doing the action, the agent should be idle state. Therefore, the gathering motion must be turned off.
- My thought is we can figure out its problem using queue_command. After every action, we can order the units to stop. Then, the agents can be idle state.

To figure out this problem, I think we need to modify Action, Agent, Core class. So, can anybody help me?

**Comment by HyeongGunLee** [comment]
I can help you on Agent part!

**Comment by gemblerz** [comment]
I think the reason why we are using `state` is because action or circumstance can continue existing. For example, having 1000 minerals goal requires gather action. and the action needs to be continued until the agent sees 1000 minerals. So, I think the agent should be able to loop the `run` function again and skip finding next action part as it continues gathering minerals.

**Comment by gemblerz** [comment]
And changing the goal state to `achieved` can make the agent's state `idle`...

**Comment by tony-sanguk** [comment]
@gemblerz Sorry, I thought that the agents can choose next action when they become idle state. I thought the agent check their knowledge base and goal only they are in idle.
I think I had better to ask how the agents work to `agent` team. Thanks!

**Comment by gemblerz** [comment]
*the agents can choose next action when they become idle state.* -- This is true.
*the agent check their knowledge base and goal only they are in idle* -- They check KB and GOAL every `tick` to go back to idle state.
Please ask us if you have any further questions and thank you for asking this.

**Comment by tony-sanguk** [comment]
@gemblerz Oh, I got it. So, is it possible to choose next action while they are working?

**Comment by gemblerz** [comment]
No. when the are working on something they just check when they can go back to idle.
Choosing next action happens only when they are idle.

**Comment by tony-sanguk** [comment]
Oops... I'm really sorry, but I'm confused...
Can I use _hangout_ later?

**Comment by gemblerz** [comment]
Sure. Don't be sorry :smile:
